### PR TITLE
Extension methods to navigate Output<Array>

### PR DESCRIPTION
**Pull request by mikhailshilkov**
An extension method seems like a less intrusive way to provide some help here. I thought of making a `OutputList` class but our `Output` is currently sealed.

### CODE REVIEW

File: sdk/dotnet/Pulumi/Core/OutputExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        /// <summary>
+        /// Convert an output containing an array to an output containing the array element
+        /// at the specified index.
+        /// </summary>
+        /// <typeparam name="T">The type of elements in the array.</typeparam>
+        /// <param name="array">An array wrapped into <see cref="Output{T}"/>.</param>
+        /// <param name="index">An index to get an element at.</param>
+        /// <returns>An <see cref="Output{T}"/> containing an array element.</returns>
+        public static Output<T> GetAt<T>(this Output<ImmutableArray<T>> array, int index)
```
Comment (CyrusNajmabadi) [inline sdk/dotnet/Pulumi/Core/OutputExtensions.cs]: this is actually a really interesting idea, and perhaps something we can do for specific packages as well.  for example, if you pull in Pulumi.Aws.S3 you'd get all the extensions to lift properties on an `Output<SomeS3Type>`.

Comment (mikhailshilkov) [reply sdk/dotnet/Pulumi/Core/OutputExtensions.cs]: Yeah, we could...
So, are you good with this API for output arrays? (this PR)

Comment (CyrusNajmabadi) [reply sdk/dotnet/Pulumi/Core/OutputExtensions.cs]: Almost.  We shoudl allow the index to be an `Input<int>`.  And we should also have a `.Length()` extension that produces an `Output<int>` imo.

Comment (mikhailshilkov) [reply sdk/dotnet/Pulumi/Core/OutputExtensions.cs]: Added

Comment (CyrusNajmabadi) [reply sdk/dotnet/Pulumi/Core/OutputExtensions.cs]: thanks!

**Pull request merged**